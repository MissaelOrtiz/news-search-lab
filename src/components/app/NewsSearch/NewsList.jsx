/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import News from './News';

const NewsList = ({ articles }) => (
  <ul aria-label="articles">
    {articles.map(({ author, title, content, urlToImage, name }) => (
      <li key={`${name}-${title}-${author}`}>
        <News author={author} title={title} content={content} urlToImage={urlToImage}/>
      </li>
    ))}
  </ul>
);

NewsList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      author: PropTypes.string,
      title: PropTypes.string,
      content: PropTypes.string.isRequired,
      urlToImage: PropTypes.string,
      name: PropTypes.string
    })
  ).isRequired,
};

export default NewsList;
