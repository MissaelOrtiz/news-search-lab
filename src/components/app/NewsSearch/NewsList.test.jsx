/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import NewsList from './NewsList';

describe('NewsList', () => {
  it('renders out a list of an article', () => {
    render(
      <NewsList articles={[
        {
          author: 'The Man Himself',
          title: 'How to Play: A comprehensive guide',
          content: 'You just gotta roll some dice and BIM BOOM BAM all done.',
          urlToImage: 'https://cdn.discordapp.com/attachments/870470157275910204/882420160567345162/20210624_003711.jpg',
          name: 'Me.'
        },
        {
          author: 'The Man Himself',
          title: 'How to Code: A comprehensive guide',
          content: 'You just gotta write some code and BIM BOOM BAM all done.',
          urlToImage: 'https://cdn.discordapp.com/attachments/870470157275910204/882420160567345162/20210624_003711.jpg',
          name: 'Me.'
        }
      ]} />
    );

    const ul = screen.getByRole('list', { name: 'articles' });
    expect(ul).toMatchSnapshot();
  });
});
