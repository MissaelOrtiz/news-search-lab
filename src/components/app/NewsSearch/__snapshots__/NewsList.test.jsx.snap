// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList renders out a list of an article 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <img
      role="img"
      src="https://cdn.discordapp.com/attachments/870470157275910204/882420160567345162/20210624_003711.jpg"
    />
    <h2>
      How to Play: A comprehensive guide
    </h2>
    <p>
       By 
      The Man Himself
       [via 
      ]
    </p>
    <p>
      You just gotta roll some dice and BIM BOOM BAM all done.
    </p>
  </li>
  <li>
    <img
      role="img"
      src="https://cdn.discordapp.com/attachments/870470157275910204/882420160567345162/20210624_003711.jpg"
    />
    <h2>
      How to Code: A comprehensive guide
    </h2>
    <p>
       By 
      The Man Himself
       [via 
      ]
    </p>
    <p>
      You just gotta write some code and BIM BOOM BAM all done.
    </p>
  </li>
</ul>
`;
